packages/client/hmi-client/src/components/model/petrinet/model-configurations/tera-stratified-value-matrix.vue

```diff
@@ -62,33 +63,24 @@
 import { ref, onMounted, computed, watch } from 'vue';
 import { cloneDeep, isEmpty } from 'lodash';
 import { StratifiedModelType } from '@/model-representation/petrinet/petrinet-service';
-import { getCatlabAMRPresentationData } from '@/model-representation/petrinet/catlab-petri';
-import {
-	getMiraAMRPresentationData,
-	getUnstratifiedParameters,
-	filterParameterLocations
-} from '@/model-representation/petrinet/mira-petri';
-import { createMatrix1D, createParameterMatrix } from '@/utils/pivot';
-import { Initial, ModelConfiguration, ModelParameter, Rate, Model } from '@/types/Types';
+import { generateMatrix } from '@/model-representation/petrinet/mira-petri';
+import { Initial, ModelConfiguration, ModelParameter, Rate } from '@/types/Types';
 import { NodeType } from '@/model-representation/petrinet/petrinet-renderer';
 import InputText from 'primevue/inputtext';
 import { pythonInstance } from '@/python/PyodideController';
 
 const props = defineProps<{
 	modelConfiguration: ModelConfiguration;
 	id: string;
+	configIndex: number;
 	stratifiedModelType: StratifiedModelType;
 	nodeType: NodeType;
 	shouldEval: boolean;
 }>();
```

Comment:
Not a fan of the addition of a `configIndex`, it makes it hard to hard to reason with the component, and more difficult to reuse because you must have some kind of array context....which is weird. Are there any blockers to just use "modelConfiguration.id" ?

Also what is props.id suppose to do?

Reply:
Decided to update the emit to take the `configIndex` by default. Updating configuration is done ultimately in `tera-model.vue` so if we want to update a config by id that'd be best done in a seperate PR where we can prepare all of its children for that. 

`props.id` is the name of the base parameter. eg. `beta`. In a transition matrix the id would just be the transition id like `t1`

Reply:
Looking at where configIndex is used (`tera-model.vue`). it's used in `updateConfiguration` to help update the frontend:

```
async function updateConfiguration(updatedConfiguration: ModelConfiguration, index: number) {
	await updateModelConfiguration(updatedConfiguration);
	setTimeout(async () => {
		emit('update-model-configuration');
		modelConfigurations.value[index] = updatedConfiguration; // Below line would be ideal but the order of the configs change after the refetch
		// await fetchConfigurations(); // elastic search might still not update in time
	}, 800);
}
```

Worth removing in a separate PR as we can just find the id

Reply:
okay - let's do a clean up pass!